cmd/mt-index-cat/main.go

```diff
@@ -0,0 +1,144 @@
+
+func main() {
+
+	var addr string
+	var from string
+	var maxAge string
+	var count bool
+
+	var total int
+
+	globalFlags := flag.NewFlagSet("global config flags", flag.ExitOnError)
+	globalFlags.StringVar(&addr, "addr", "http://localhost:6060", "graphite/metrictank address")
+	globalFlags.StringVar(&from, "from", "30min", "from. eg '30min', '5h', '14d', etc. or a unix timestamp")
+	globalFlags.StringVar(&maxAge, "max-age", "6h30min", "max age (last update diff with now) of metricdefs.  use 0 to disable")
+	globalFlags.BoolVar(&count, "count", false, "print number of metrics loaded to stderr")
+
+	cassFlags := cassandra.ConfigSetup()
+
+	outputs := []string{"dump", "list", "vegeta-render", "vegeta-render-patterns"}
+
+	flag.Usage = func() {
+		fmt.Println(os.Args[0])
+		fmt.Printf("Usage:\n\n")
+		fmt.Printf("  inspect-idx [global config flags] <idxtype> [idx config flags] output \n\n")
+		fmt.Printf("global config flags:\n\n")
+		globalFlags.PrintDefaults()
+		fmt.Println()
+		fmt.Printf("idxtype: only 'cass' supported for now\n\n")
+		fmt.Printf("cass config flags:\n\n")
+		cassFlags.PrintDefaults()
+		fmt.Println()
+		fmt.Printf("output: %v\n\n\n", strings.Join(outputs, "|"))
+		fmt.Println("EXAMPLES:")
+		fmt.Println("mt-index-cat -from 60min cass -hosts cassandra:9042 -keyspace raintank list")
+	}
+
+	if len(os.Args) == 2 && (os.Args[1] == "-h" || os.Args[1] == "--help") {
+		flag.Usage()
+		os.Exit(0)
+	}
+
+	if len(os.Args) < 3 {
+		flag.Usage()
+		os.Exit(-1)
+	}
+
+	var show func(d schema.MetricDefinition)
+
+	last := os.Args[len(os.Args)-1]
+	var found bool
+	for _, output := range outputs {
+		if last == output {
+			found = true
+		}
+	}
+	if !found {
+		log.Printf("invalid output %q", last)
+		flag.Usage()
+		os.Exit(-1)
+	}
+	var cassI int
+	for i, v := range os.Args {
+		if v == "cass" {
+			cassI = i
+		}
+	}
+	if cassI == 0 {
+		log.Println("only indextype 'cass' supported")
+		flag.Usage()
+		os.Exit(1)
+	}
+
+	globalFlags.Parse(os.Args[1:cassI])
+	cassFlags.Parse(os.Args[cassI+1 : len(os.Args)-1])
+	cassandra.Enabled = true
+
+	switch os.Args[len(os.Args)-1] {
+	case "dump":
+		show = out.Dump
+	case "list":
+		show = out.List
+	case "vegeta-render":
+		show = out.GetVegetaRender(addr, from)
+	case "vegeta-render-patterns":
+		show = out.GetVegetaRenderPattern(addr, from)
+	default:
+		panic("this should never happen. we already validated the output type")
+	}
+
+	idx := cassandra.New()
+	err := idx.InitBare()
+	perror(err)
+
+	// from should either be a unix timestamp, or a specification that graphite/metrictank will recognize.
+	_, err = strconv.Atoi(from)
+	if err != nil {
+		_, err = dur.ParseUNsec(from)
+		perror(err)
+	}
+
+	maxAgeInt := int64(0)
+	if maxAge != "0" {
+		var i uint32
+		i, err = dur.ParseUNsec(maxAge)
+		perror(err)
+		maxAgeInt = int64(i)
+	}
+
+	var defs []schema.MetricDefinition
+	defs = idx.Load(defs)
+
+	for _, d := range defs {
+		if maxAgeInt != 0 && d.LastUpdate > time.Now().Unix()-maxAgeInt {
```

Comment:
actually this made me realize there's a bug here. if maxAgeInt is 0, it means we should show them all without filtering.